plugins/PrivacyManager/Tasks.php

```diff
@@ -8,13 +8,39 @@
  */
 namespace Piwik\Plugins\PrivacyManager;
 
+use Piwik\Plugins\PrivacyManager\Model\LogDataAnonymizations;
+
 class Tasks extends \Piwik\Plugin\Tasks
 {
 
+    /**
+     * @var LogDataAnonymizations
+     */
+    private $logDataAnonymizations;
+
+    public function __construct(LogDataAnonymizations $logDataAnonymizations)
+    {
+        $this->logDataAnonymizations = $logDataAnonymizations;
+    }
+
     public function schedule()
     {
         $this->daily('deleteReportData', null, self::LOW_PRIORITY);
         $this->daily('deleteLogData', null, self::LOW_PRIORITY);
+        $this->hourly('anonymizePastData', null, self::LOW_PRIORITY);
+    }
+
+    public function anonymizePastData()
+    {
+        $loop = 0;
+        do {
+            $loop++; // safety loop...
+            $id = $this->logDataAnonymizations->getNextScheduledAnonymizationId();
+            if (!empty($id)) {
+                $this->logDataAnonymizations->executeScheduledEntry($id);
+            }
+
+        } while (!empty($id) && $loop < 100);
```

Comment:
What's the loop for? To reduce the chance of missing an entry that was just created? Or to execute the first 100 entries?

Reply:
I do this with most `do/while` loops just to prevent to not run forever in case the application is somehow buggy and would always return an `$id`. Then it would run forever.

Reply:
I see, it's a `while (true)` loop, w/ a safety counter to make sure it doesn't go forever! Thought the 100 was important or something :)